lib/cog/models/relay.ex

```diff
@@ -24,11 +24,24 @@ defmodule Cog.Models.Relay do
     model
     |> Repo.preload(:groups)
     |> cast(params, @required_fields, @optional_fields)
+    |> allow_user_defined_id_on_insert(params)
     |> validate_presence_on_insert(:token)
     |> encode_token
     |> unique_constraint(:name)
   end
 
+  def allow_user_defined_id_on_insert(changeset, params) do
+    case {get_field(changeset, :id), Map.get(params, "id")} do
```

Comment:
Ah good point. `@optional_fields` is poorly named. It really means `@whitelisted_fields`. Another reason to switch was to avoid having to check for both `:id` and `"id"` in params. I pushed the change so I'll just merge that. This will change with Ecto 2.0 anyway since they handle whitelisting and requiring params differently.